pkg/kubectl/conditions.go

```diff
@@ -117,38 +79,21 @@ func PodRunningAndReady(event watch.Event) (bool, error) {
 		return false, errors.NewNotFound(schema.GroupResource{Resource: "pods"}, "")
 	}
 	switch t := event.Object.(type) {
-	case *api.Pod:
-		switch t.Status.Phase {
-		case api.PodFailed, api.PodSucceeded:
-			return false, ErrPodCompleted
-		case api.PodRunning:
-			return pod.IsPodReady(t), nil
-		}
 	case *corev1.Pod:
```

Comment:
How do you know it's never going to be a `api.Pod`?

Reply:
Both "PodCompleted" and "PodRunningAndReady" functions are only called once each in run.go. In fact, they are only called as an argument in this file: "waitForPod".  Within "waitForPod", the functions are called, and only external versions of Pod are used.